app/views/reports/regions/_visit_details_card.html.erb

```diff
@@ -26,23 +26,23 @@
                   <span data-missed-visits-patients="<%= number_with_delimiter(@data.last_value(:missed_visits)) %>">
                     <%= number_with_delimiter(@data.last_value(:missed_visits)) %>
                   </span>
-                  patients with no visit from 
-                  <span data-period-start="<%= @data.last_value(:period_info)["bp_control_start_date"] %>">
-                    <%= @data.last_value(:period_info)["bp_control_start_date"] %>
+                  patients with no visit from
+                  <span data-period-start="<%= @period.bp_control_range_start_date %>">
+                    <%= @period.bp_control_range_start_date %>
```

Comment:
Instead of relying on the values in the `@data`, we can grab all these dates from the `@period` itself. We know the controller will set that period based on either the default current period (ie current month), or the period the user selects. So this is safer and more explicit.